pkg/cmd/roachtest/tests/online_restore.go

```diff
@@ -242,6 +194,104 @@ func registerOnlineRestore(r registry.Registry) {
 	}
 }
 
+func registerOnlineRestoreCorrectness(r registry.Registry) {
+	sp := restoreSpecs{
+		hardware: makeHardwareSpecs(hardwareSpecs{workloadNode: true}),
+		backup: makeRestoringBackupSpecs(backupSpecs{
+			nonRevisionHistory: true,
+			version:            fixtureFromMasterVersion,
+			workload:           tpccRestore{opts: tpccRestoreOptions{warehouses: 10, workers: 1, waitFraction: 0}}}),
+		timeout:                15 * time.Minute,
+		suites:                 registry.Suites(registry.Nightly),
+		restoreUptoIncremental: 1,
+		namePrefix:             "online/correctness",
+	}
+	sp.initTestName()
+	r.Add(
+		registry.TestSpec{
+			Name:                sp.testName,
+			Owner:               registry.OwnerDisasterRecovery,
+			Cluster:             sp.hardware.makeClusterSpecs(r, sp.backup.cloud),
+			Timeout:             sp.timeout,
+			CompatibleClouds:    sp.backup.CompatibleClouds(),
+			Suites:              sp.suites,
+			SkipPostValidations: registry.PostValidationReplicaDivergence,
+			Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+				tpccWorkload, ok := sp.backup.workload.(tpccRestore)
+				if !ok {
+					require.Fail(t, "only tpcc workloads are supported for correctness testing")
+				}
+				regRestoreSpecs := sp
+				regRestoreSpecs.timeout = time.Duration(sp.timeout.Seconds()*0.45) * time.Second
+				regTpccWorkload := tpccWorkload
+				regTpccWorkload.opts.queryTraceFile = strings.ReplaceAll(sp.testName, "/", "-") + "-reg.trace"
+				if regTpccWorkload.opts.seed == 0 {
+					regTpccWorkload.opts.seed = crdbworkload.NewUint64RandomSeed().Seed()
+				}
+				if regTpccWorkload.opts.fakeTime == 0 {
+					regTpccWorkload.opts.fakeTime = uint32(timeutil.Now().Unix())
+				}
+				regRestoreSpecs.backup.workload = regTpccWorkload
+				regRestoreHook := restoreHook{
+					sp:          regRestoreSpecs,
+					runOnline:   false,
+					runWorkload: true,
+				}
+
+				onlineRestoreSpecs := sp
+				onlineRestoreSpecs.timeout = regRestoreSpecs.timeout
+				onlineTpccWorkload := tpccWorkload
+				onlineTpccWorkload.opts.queryTraceFile = strings.ReplaceAll(sp.testName, "/", "-") + "-online.trace"
```

Comment:
nit: there's some repeat code here and above. could you create a helper to make this cleaner?